src/atoms/Icon.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { Component } from 'react'
+import {
+    findIconDefinition,
+    IconDefinition,
+    IconLookup,
+    IconName,
+    IconPrefix,
+    library
+} from '@fortawesome/fontawesome-svg-core'
+import { fab } from '@fortawesome/free-brands-svg-icons'
+import { far } from '@fortawesome/free-regular-svg-icons'
+import { fas } from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { Styles } from '../constants/enum'
+
+library.add(fab, far, fas)
+
+interface IconProps {
+    className?: string
+    name: IconName
+    style?: Styles
+}
+
+class Icon extends Component<IconProps> {
+    public static defaultProps = {
+        className: '',
+        style: Styles.Solid
+    }
+
+    constructor (props: Readonly<IconProps>) {
+        super(props)
+
+        this.getIconDefinition = this.getIconDefinition.bind(this)
+        this.getIconPrefix = this.getIconPrefix.bind(this)

```

Comment:
This line is useless. The goal of this is to bind the class instance (this) to the methods we will call in the JSX of the render method. In you JSX, you only call getIconDefinition and not getIconPrefix :)

Reply:
Done.